### PR TITLE
Melinda Deinert

### PR DESCRIPTION
**Pull request by mel-d**

### CODE REVIEW

**Comment by mel-d** [comment]
My Google spreadsheet: https://docs.google.com/spreadsheets/d/1idJiZymJbdUrKtmJxX8I-KH3dZ2AvxnZrd5KQ8c-FrI/edit?usp=sharing

**Comment by mel-d** [comment]
If I'm heading in the right direction, the plan is to make a table for each event. That's why I'm thinking I'll either need to delete some of the events or add a whole boatload of competitors--is this ok on a smaller scope, or do you want huge amounts of events + competitors to make it realistic?

**Comment by sumeetjain** [comment]
I suggest:
- Creating a table for "styles", and then having each event refer to a style_id
- Creating a table for "universities", and then having each student refer to a university_id
- Creating a table for "conferences", and then having each university refer to a conference_id

Definitely do not make a table for each event. Consider instead a table like "event_results" or "event_performances", which contain references to an "event_id".

**Comment by mel-d** [comment]
Ok, I fulfilled those three bullet points and finished filling in result times. I don't want my brain to be smashed when I walk in tomorrow morning, so I think I'm going to hold off until class tomorrow to finish the part 2 of phase 1.

**Comment by sumeetjain** [comment]
Everything looks great. Just want to make sure you're aware:



is not possible. You did that just for the sake of cleanliness, right?

In the actual database each row will have its own event_id value.

File: app/models/models.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Event < ActiveRecord::Base
+end
+
+class Style <ActiveRecord::Base
+end
```
Comment (sumeetjain) [inline app/models/models.rb]: While it works to put models in the same file, please separate them out into their own files.

Comment (sumeetjain) [inline app/models/models.rb]: It'll be better later when we start doing more complicated things with them.

File: db/define_schema.rb [hunk 2]
```diff
@@ -11,4 +11,13 @@
...
 
 # What table do you need to store the necessary information?
-# What columns do you need in that table?
\ No newline at end of file
+# What columns do you need in that table?
+
+DB.define_table("events")
+DB.define_column("events", "gender", "string")
+DB.define_column("events", "distance", "string")
+DB.define_column("events", "style id", "integer")
```
Comment (sumeetjain) [inline db/define_schema.rb]: Column names cannot have spaces. This should be `style_id`.

**Comment by sumeetjain** [comment]
The rest of Phase 1, Part 2 is to build the action controller actions and views that are needed to create new rows in the relevant tables.

**Comment by mel-d** [comment]
I did things a little out of order because I wanted to make progress before asking you another question-- I think all the schema should be there, and I've made one view for each table, but there's nothing on them yet. What I want to clarify is re: the input we're designing on the front page of the swim meet--is it supposed to be designed for an athlete registering for the meet, or for an administrator who is inputting results after a race has finished?

**Comment by sumeetjain** [comment]
All pages should be designed for the administrator only. No one else will use the system.

**Comment by mel-d** [comment]
I just wanted to check-in again to make sure I'm going about this correctly. The first form on the first page is functional in that it leads to another page, but there's no database yet.

**Comment by sumeetjain** [comment]
Direction looks good so far.

**Comment by mel-d** [comment]
Ok, so I've got the forms, views, and links etc. working. I'm just now starting my attempt at getting a working database. When I run rackup and visit the /results page, I get this error:

ArgumentError at /results
wrong number of arguments (given 1, expected 0)

```
file: results_controller.rb location: block in <top (required)> line: 7 
```

**Comment by mel-d** [comment]
Is it stemming from me having somehow done the @results = Result.all on the results_controller file? I defined a class called Result in the models.

**Comment by mel-d** [comment]
I meant to say "is it stemming form me having done the @results = Result.all wrong? Sentence-forming fail.

**Comment by StevenThiesfeld** [comment]
It looks like you're defining @variables in the '/results_added' controller that you want in your 'results' controller, like @results, @events etc.

**Comment by mel-d** [comment]
I was wondering about that--but if I define the variables in the /results controller (and I assume I'd be moving pretty much everything up to the other controller?) wouldn't it only save to the database when someone goes to the /results page, as opposed to when they actually submit through the form and wind up at the results_added page?

**Comment by mel-d** [comment]
Just for the record, I did just move those up to the /results controller and got the same error message.

**Comment by mel-d** [comment]
Ok, sorry to be updating you as you're probably answering my last question, but I moved the @ variable definitions to /results and left the x.whatever in results_added. The /results page seems to be working now, so I think I just need to figure out what's going wrong with actually adding the data.

**Comment by StevenThiesfeld** [comment]
Just so I understand your question, what's the error message (if any) you're getting now?

**Comment by mel-d** [comment]
Sorry--I'm all over the place right now. Here's my newest error message:

ArgumentError at /result_added
wrong number of arguments (given 1, expected 0)

```
file: results_controller.rb location: block in <top (required)> line: 11 
```

And this comes when I try to enter info into the results form and submit. Is there supposed to be a parameter when I declare x = Result.new or something?

**Comment by StevenThiesfeld** [comment]
Hmm, I still don't see anything that would throw an error

**Comment by StevenThiesfeld** [comment]
can you push up your latest change?

**Comment by mel-d** [comment]
Done!

**Comment by mel-d** [comment]
I figured this one out--it was because I was missing some stuff in my define_schema file. I'm having different issues now, but I'll try to figure them out myself for a bit before I bug you about it. Thanks!

**Comment by mel-d** [comment]
I guess I mistitled this push a little in that I think the database is actually working (the adding results one; as that's the only one I've tried to make save thus far), I'm just not able to display it. On the /results page, which should display all stored results, I only see this: 

finished in # seconds. finished in # seconds. finished in # seconds.

**Comment by StevenThiesfeld** [comment]
in your results.erb file in the each loops you're displaying the object itself, you need to specify what you want to see (i.e e.id or r.time)

**Comment by mel-d** [comment]
I just figured that out too, and it's working now! Thanks! I'm a little embarrassed that it took me hours to figure that out.

**Comment by mel-d** [comment]
I'm getting nearly everything I've tried today to work--my main issue has been displaying the values chosen when the admin picks from a drop-down box instead of typing in a value (specifically the info about competitors)--is there something I'm doing wrong on my competitors.erb page?

**Comment by mel-d** [comment]
(I have tried Googling; I just keep finding different tutorials to actually create the drop-down list, not how to display its selected value via html.)

**Comment by sumeetjain** [comment]
What have you tried so far?

**Comment by mel-d** [comment]
I originally had those two values set as integers that'd display the school id and conference id, but if there's a drop-down list it'd might as well display the actual names. So after I switched them both to strings and using _name instead of _id, I've been stuck.

**Comment by mel-d** [comment]
Acccckk, it's because I never changed the names on the controller. Of course I found it two minutes after I messaged you instead of two minutes before.

**Comment by mel-d** [comment]
Very confused about why my display_competitor_first_name is returning an error. It says undefined method 'first_name' for nil, but I can't understand why y would be nil if it's part of an each loop to go through each result on the results.erb page.

**Comment by sumeetjain** [comment]
It's saying `first_name` is nil. What file and line?

**Comment by mel-d** [comment]
result_model.rb, line 5

**Comment by mel-d** [comment]
I've gotten the method display_conference_of_school to work (it's on school_model.rb), but for some reason when I try to loop through each school on schools.erb, it only picks up this method the first time (it does pick up on every other part of the loop each time it iterates). The [0] after the return array is just so that the name of the conference will only appear one time even if more than one competitor is from there.

**Comment by mel-d** [comment]
(because the return array would have the name of the conference once for each competitor from there)

**Comment by sumeetjain** [comment]
I'm somewhat confused by that method.

Why is it calling the `Competitor` model at all?

Can you describe the algorithm you're trying to build _without any code_ first?

**Comment by mel-d** [comment]
Yes--It's calling the competitor model because that's the only table that has both the university and the conference listed together. (So I'm trying to do what you showed us this morning accessing information through more than one table).. so this method should take the school name, find the competitors who go to that school, and then return the name of their conference.

**Comment by mel-d** [comment]
Just for the record, when I run binding.pry it finds the correct x every time. It only finds y the first time and then returns nil in every other iteration.

**Comment by mel-d** [comment]
IT WAS A FUGGIN DATA MISHAP. The values on the drop-down boxes not being exactly the names of the universities within the tables threw it off--When I changed the spellings to match exactly, I didn't realize none of the previously-added data was being updated.

**Comment by sumeetjain** [comment]
Doh. Glad you figured it out!

**Pull request closed without merging**